### PR TITLE
add a way to listen to both http and https

### PR DESCRIPTION
**Pull request by kahing**

### CODE REVIEW

File: src/main/java/org/gaul/s3proxy/Main.java [hunk 1]
```diff
@@ -57,10 +57,12 @@ public static void main(String[] args) throws Exception {
...
+        String secureEndpoint = properties.getProperty(
+                S3ProxyConstants.PROPERTY_SECURE_ENDPOINT);
         String s3ProxyAuthorization = properties.getProperty(
                 S3ProxyConstants.PROPERTY_AUTHORIZATION);
         if (provider == null || identity == null || credential == null ||
-                s3ProxyEndpointString == null ||
+                (s3ProxyEndpointString == null && secureEndpoint == null) ||
                 s3ProxyAuthorization == null) {
             System.err.println("Properties file must contain:\n" +
                     Constants.PROPERTY_PROVIDER + "\n" +
```
Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/Main.java]: Update error message?

File: src/main/java/org/gaul/s3proxy/S3Proxy.java [hunk 2]
```diff
@@ -144,6 +162,10 @@ public int getPort() {
         return ((ServerConnector) server.getConnectors()[0]).getLocalPort();
     }
 
+    public int getSecurePort() {
+        return ((ServerConnector) server.getConnectors()[1]).getLocalPort();
+    }
+
```
Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/S3Proxy.java]: Does this throw `ArrayOutOfBoundException` when a user only configured HTTPS?

**Comment by andrewgaul** [comment]
Why do the integration tests fail?

**Comment by kahing** [comment]
does travis generate the keys in keystore? it's complaining that keystore doesn't exist

**Comment by andrewgaul** [comment]
`run-s3-tests.sh` only configures HTTP; can we just disable the secure endpoint for this test?

**Comment by andrewgaul** [comment]
Rebased and pushed to master as 6e4ebad1c8982cc0259fbdafedb4434ef6955119, f17d968a3341a80a7d80e12aa56b16f76d3c4604, and 170aedb319cc1b9b692d00b93d1a946cb876586a.

**Pull request closed without merging**